superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -415,29 +557,20 @@ class AnnotationLayer extends React.PureComponent {
     this.props.close();
   }
 
-  renderOption(option) {
+  renderChartHeader(label, description, value) {
     return (
-      <span
-        css={{
-          overflow: 'hidden',
-          textOverflow: 'ellipsis',
-          whiteSpace: 'nowrap',
-        }}
-        title={option.label}
-      >
-        {option.label}
-      </span>
+      <ControlHeader
+        hovered
+        label={label}
+        description={description}
+        validationErrors={!value ? ['Mandatory'] : []}
+      />
     );
   }
 
   renderValueConfiguration() {
-    const {
-      annotationType,
-      sourceType,
-      value,
-      valueOptions,
-      isLoadingOptions,
-    } = this.state;
+    const { annotationType, sourceType, value, valueOptions } = this.state;
+    const validValue = valueOptions?.[value] ?? null;
```

Comment:
`valueOptions` is just being read here to get the actual `value`. Your state already have a `value` property so I think you can remove all references to `valueOptions`.

Reply:
I'm using 'valueOptions' to store the object passed to `handleSelectValue` which contains the id and label. This addresses a bug in which the dropdown field appears blank after first selecting/mounting a saved annotation when feeding it the id alone (the `value` in state).

Reply:
Why don't you change the `value` to be an object with `id` and `label`?

Reply:
`value` has to be kept strictly as the id to suit the way the annotation is being applied to the chart.

Reply:
I'm not sure if I understood. In places where you need the id, you can use `value.id` and in places where you need the id/label pair, you can use `value`. Is there something I'm missing?

Reply:
I'm thinking of the `applyAnnotation` function where the `value` is being taken from state and can only be the id of the native annotation or chart in order for the annotation to be applied in `AnnotationLayerControl`. If it contains an object the annotation errors. I ran into this when trying to get rid of valueOptions much earlier on and thought best to leave `value` as is to avoid increasing the scope of the changes. Looking at it now, I could change `newAnnotation.value` to be `value.id` if you think that's a better idea?

Reply:
I think you can set

```
newAnnotation.id = value.id;
```

The same way you manually set `color`. You just need to remove `value` from `annotationFields`. What I'm trying to achieve is to simplify the state where the value is set in just one place.

Reply:
Pushed a commit to address this, let me know what you think. Added a comment to explain the change in type of `value`